pkg/cloudprovider/providers/gce/gce_loadbalancer.go

```diff
@@ -875,6 +937,44 @@ func (gce *GCECloud) firewallNeedsUpdate(name, serviceName, region, ipAddress st
 	return true, false, nil
 }
 
+func (gce *GCECloud) ensureHttpHealthCheckFirewall(region string, hosts []*gceInstance, hcName string, hcPort int32, isNodesHealthCheck bool) error {
+	// Prepare the firewall params for creating / checking.
+	desc := "Firewall for health checks from load-balancer."
+	sourceRanges := lbSrcRngsFlag.ipn
+	ports := []v1.ServicePort{{Protocol: "tcp", Port: hcPort}}
+
+	clusterId, err := gce.ClusterId.GetId()
+	if err != nil {
+		return fmt.Errorf("error getting cluster ID: %v", err)
+	}
+	fwNameSuffix := MakeHealthCheckFirewallNameSuffix(clusterId, hcName, isNodesHealthCheck)
+	fwName := makeFirewallName(fwNameSuffix)
```

Comment:
I think it makes sense to let existing resources (http healthcheck for nodeport, targetpool, forwarding rule, firewall rule for ELB svc) keep the old name now, and we can deal with the renaming/migration later if needed?